## Goals

* Optimize GFF format definiation and usability
* Detect methodology accuracy due to tools and some experimental step in the protocols.

## Data

http://www.biorxiv.org/content/biorxiv/early/2017/05/17/113050.full.pdf

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?token=whipakmajrwprcv&acc=GSE94586

## Tools

* bcbio smallRNA-seq pipeline + isomiRs - On charge Lorena Pantano
* isomiR-SEA - On charge Gianvito Urgese
* ChimiRa, miRge - On charge Marck Halushka
* sRNAbench - On charge Michael Hackenberg
* Prost - Thomas Desvignes
* miRGe - Marc K. Halushka
* [Add your tool here and person will do it]

## Questions to address

* [To be added]

## Milestones:

### Set up

* [X] Select random public data
* [X] Run with all the tools listed above
* [X] Put data in common space
* [Working] Adapt output tools to GFF format

### Random sample

Sample [SRR5756178](https://www.ncbi.nlm.nih.gov/sra/?term=SRR5756178) is a whole blood small RNA-seq run from this manuscript https://academic.oup.com/nar/article/4080663 and is part of project PRJNA391912.  It has ~ 2.8 million reads, of which ~2.6 million are miRNAs.

### [To be added]
